test/widgets/actions_test.dart

```diff
@@ -188,95 +302,23 @@ void main() {
-        ..bodyFields.deepEquals({
-            'stream_id': stream.streamId.toString(),
-          });
-    });
-
-    testWidgets('TopicNarrow on legacy server', (WidgetTester tester) async {
-      final narrow = TopicNarrow.ofMessage(eg.streamMessage());
-      await prepare(tester);
-      connection.zulipFeatureLevel = 154;
-      connection.prepare(json: {});
-      markNarrowAsRead(context, narrow, true); // TODO move legacy-server check inside markNarrowAsRead
-      await tester.pump(Duration.zero);
-      check(connection.lastRequest).isA<http.Request>()
-        ..method.equals('POST')
-        ..url.path.equals('/api/v1/mark_topic_as_read')
-        ..bodyFields.deepEquals({
-            'stream_id': narrow.streamId.toString(),
-            'topic_name': narrow.topic,
-          });
-    });
-
-    testWidgets('DmNarrow on legacy server', (WidgetTester tester) async {
-      final message = eg.dmMessage(from: eg.otherUser, to: [eg.selfUser]);
-      final narrow = DmNarrow.ofMessage(message, selfUserId: eg.selfUser.userId);
-      final unreadMsgs = eg.unreadMsgs(dms: [
-        UnreadDmSnapshot(otherUserId: eg.otherUser.userId,
-          unreadMessageIds: [message.id]),
-      ]);
-      await prepare(tester, unreadMsgs: unreadMsgs);
-      connection.zulipFeatureLevel = 154;
-      connection.prepare(json:
-        UpdateMessageFlagsResult(messages: [message.id]).toJson());
-      markNarrowAsRead(context, narrow, true); // TODO move legacy-server check inside markNarrowAsRead
-      await tester.pump(Duration.zero);
-      check(connection.lastRequest).isA<http.Request>()
-        ..method.equals('POST')
-        ..url.path.equals('/api/v1/messages/flags')
-        ..bodyFields.deepEquals({
-            'messages': jsonEncode([message.id]),
-            'op': 'add',
-            'flag': 'read',
-          });
-    });
-
     testWidgets('catch-all api errors', (WidgetTester tester) async {
-      final zulipLocalizations = GlobalLocalizations.zulipLocalizations;
       const narrow = CombinedFeedNarrow();
+      final apiNarrow = narrow.apiEncode()..add(ApiNarrowIsUnread());
       await prepare(tester);
       connection.prepare(exception: http.ClientException('Oops'));
-      markNarrowAsRead(context, narrow, false);
+      updateMessageFlagsStartingFromAnchor(
+        context: context, apiNarrow: apiNarrow, op: UpdateMessageFlagsOp.add, flag: MessageFlag.read,
+        includeAnchor: false, startAnchor: AnchorCode.oldest, onCompletedMessage: onCompletedMessage,
+        onFailedTitle: onFailedTitle, onProgressTitle: onProgressTitle);
       await tester.pump(Duration.zero);
       await tester.pumpAndSettle();
       checkErrorDialog(tester,
-        expectedTitle: zulipLocalizations.errorMarkAsReadFailedTitle,
+        expectedTitle: onFailedTitle,
         expectedMessage: 'NetworkException: Oops (ClientException: Oops)');
     }, skip: true, // TODO move this functionality inside markNarrowAsRead
```

Comment:
In the first commit, we refactored `markNarrowAsRead` to handle this TODO. We should update this test in that commit and possibly not skip it.